src/System.Reflection/tests/MethodInfo/MethodInfo_InvokeTests.cs

```diff
@@ -4,252 +4,386 @@
         {
-            MethodInfo mi = null;
-            int retVal = 0;
-            int expectedVal = 110;
             object[] varParams =
             {
-                ( 10 ),
-                ( 100 )
+                10,
+                100
             };
 
-            mi = getMethod(typeof(Co1333Invoke), "ReturnI4Sum");
-            retVal = (int)mi.Invoke(null, varParams);
-
-            Assert.True(retVal.Equals(expectedVal), string.Format("Failed! MethodInfo.Invoke did not return correct result. Expected {0} , Got {1}", expectedVal, retVal));
+            Assert.Equal(110, (int)getMethod(typeof(Co1333Invoke), "ReturnI4Sum").Invoke(null, varParams));
         }
 
+        // Invoke a static method using class object, that has parameters of primitive types
+        [Fact]
+        public static void TestInvokeMethod_StaticOnInstanceWithPrimitiveParameters()
+        {
+            Co1333Invoke clsObj = new Co1333Invoke();
+            object[] varParams =
+            {
+                10,
+                100
+            };
 
+            Assert.Equal(110, (int)getMethod(typeof(Co1333Invoke), "ReturnI4Sum").Invoke(clsObj, varParams));
+        }
 
-        //Invoke a static method using class object , that has parameters of primitive types
+        // Invoke inherited static method.
         [Fact]
-        public static void TestInvokeMethod8()
+        public static void TestInvokeMethod_InheritedStaticOnInstance()
         {
-            MethodInfo mi = null;
             Co1333Invoke clsObj = new Co1333Invoke();
-            int retVal = 0;
-            int expectedVal = 110;
             object[] varParams =
             {
-                ( 10 ),
-                ( 100 )
+                10,
             };
 
-            mi = getMethod(typeof(Co1333Invoke), "ReturnI4Sum");
-            retVal = (int)mi.Invoke(clsObj, varParams);
-
-            Assert.True(retVal.Equals(expectedVal), string.Format("Failed! MethodInfo.Invoke did not return correct result. Expected {0} , Got {1}", expectedVal, retVal));
+            Assert.Equal(true, (bool)getMethod(typeof(Co1333_a), "IsEvenStatic").Invoke(clsObj, varParams));
         }
 
-
-        //Invoke inherited static method.
+        // Enum reflection in method signature
         [Fact]
-        public static void TestInvokeMethod9()
+        public static void TestInvokeMethod_EnumParameterAndEnumReturnValue()
         {
-            MethodInfo mi = null;
             Co1333Invoke clsObj = new Co1333Invoke();
-            bool retVal = false;
-            bool expectedVal = true;
-            object[] varParams =
+            object[] vars =
             {
-                ( 10 ),
+                MyColorEnum.RED
             };
 
-            mi = getMethod(typeof(Co1333_a), "IsEvenStatic");
-            retVal = (bool)mi.Invoke(clsObj, varParams);
+            Assert.Equal((int)MyColorEnum.GREEN, (int)getMethod(typeof(Co1333Invoke), "GetAndRetMyEnum").Invoke(clsObj, vars));
```

Comment:
Nit: Rather than casting both the expected and actual values to int, could you just cast the expectd value to MyColorEnum?